secret_store/src/key_server_cluster/io/deadline.rs

```diff
@@ -16,18 +16,24 @@
 
 use std::io;
 use std::time::Duration;
-use futures::{Future, Select, Poll, Async};
-use tokio_core::reactor::{Handle, Timeout};
+use futures::{Future, Poll, Async};
+use tokio::timer::Timeout;
 
 type DeadlineBox<F> = Box<Future<Item = DeadlineStatus<<F as Future>::Item>, Error = <F as Future>::Error> + Send>;
 
 /// Complete a passed future or fail if it is not completed within timeout.
-pub fn deadline<F, T>(duration: Duration, handle: &Handle, future: F) -> Result<Deadline<F>, io::Error>
-	where F: Future<Item = T, Error = io::Error> + Send + 'static, T: 'static {
-	let timeout: DeadlineBox<F> = Box::new(Timeout::new(duration, handle)?.map(|_| DeadlineStatus::Timeout));
-	let future: DeadlineBox<F> = Box::new(future.map(DeadlineStatus::Meet));
+pub fn deadline<F, T>(duration: Duration, future: F) -> Result<Deadline<F>, io::Error>
```

Comment:
Seems like logic has changed here: previously it was:
future completes with Ok before timeout => Ok(DeadlineStatus::Meet)
future completes with  Err before timeout => Err
future doesn't complete before timeout => Ok(DeadlineStatus::Timeout)

Now it always returns Ok

Reply:
Yeah I probably did that as a temporary measure and forgot to go back and correct it. Double-check my corrections to make sure it looks ok.